pgxn/neon/pagestore_smgr.c

```diff
@@ -1278,16 +1278,20 @@ static void neon_read_at_lsn_multi_region(RelFileNode rnode, ForkNumber forkNum,
 	{
 		case T_NeonGetPageResponse:
 			memcpy(buffer, ((NeonGetPageResponse *) resp)->page, BLCKSZ);
-			if (RegionIsRemote(region))
-			{
-				XLogRecPtr lsn = ((NeonGetPageResponse *) resp)->lsn;
-				/*
-					Set the LSN on the page to be equal to the LSN snapshot of the current transaction
-					so that we don't need to evict the page from local buffer if we use the same LSN
-					snapshot for the subsequent transactions.
-				*/
-				PageSetLSN((Page) buffer, lsn);
-			}
+			// TODO (ctring): potential optimization, but need more testing before uncommenting
```

Comment:
I had an error due to [this check](https://github.com/DSLAM-UMD/postgres/blob/ctring-update/src/backend/storage/buffer/bufmgr.c#L1040-L1059). 

What happened was that when I inserted new tuples, postgres tried to get the last page of the relation. This page might not exist previously (for example, when the relation is completely new), so postgres would expect that this page was either valid (with correctly set up header) or completely blank. Since the page did not exist, neon correctly returned the all-zero page. However, because we always set the LSN on the page here, it failed the all-zero check.

What this code should do instead is only update the LSN if the LSN is not 0. I already made this fix, but I wasn't sure what other problems it might cause so I decided to leave it there and come back later. It shouldn't affect correctness because the page would still present throughout the transaction, it just cannot be reused across different transactions in the same session.